keps/sig-auth/4412-projected-service-account-tokens-for-kubelet-image-credential-providers/README.md

```diff
@@ -0,0 +1,1123 @@
+be configurable in the kubelet credential provider configuration.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a Kubernetes administrator, I need to securely manage image pulls for my multi-tenant
+cluster so that I can prevent unauthorized access to private container images and reduce
+the security risks associated with static secrets.
+
+#### Story 2
+
+As a developer deploying applications in Kubernetes, I want my pods to use dynamic
+service account tokens to pull images securely, ensuring that each pod only accesses images
+that it is explicitly permitted to use, thereby maintaining proper segmentation of access
+between services.
+
+#### Story 3
+
+As a security engineer, I need to eliminate the use of static secrets for image pulls
+and leverage workload identity to enhance security through ephemeral, tightly scoped tokens.
+This change reduces the attack surface and the risk of credential leaks, aligning with
+best practices for security.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+To keep this KEP small both in terms of implementation and scope,
+the design is written to meet the following criteria:
+
+- No changes to any Kubernetes REST APIs
+- No changes to the CRI API
+- No changes to how the Kubelet interacts with the CRI API
+- No changes to any registry
+- Minimal changes to Kubelet
+- Minimal changes to Kubelet credential providers
+- Minimal changes to Kube API Server
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+#### Credential Provider Configuration
+
+Adding a new field `TokenAttributes` to the `CredentialProvider` struct in the kubelet configuration.
+
+1. `ServiceAccountTokenAudience` is the intended audience for the credentials. If set, the kubelet will generate a service account token for the audience and pass it to the plugin.
```

Comment:
before we never sent any secrets to the credential provider. So it was never required to be secured strictly to avoid the attack when it is replaced with some other binary. With this change we make this attack meaningful. Do we have enough documentation suggesting how to secure the cred provider so it cannot be replaced?

Reply:
I don't see any documentation [here](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-credential-provider/) but we can certainly get that added as part of the implementation.

Reply:
The credential provider is executed as the same user as the kubelet, so probably root, meaning it can trivially see the kubelet creds from the filesystem and have access to anything that the kubelet does (including tokens for pods).